tests/unit/test_bayesian_optimizer.py

```diff
@@ -259,7 +259,7 @@ def __deepcopy__(self, memo: dict[int, object]) -> _UncopyableModel:
     with pytest.raises(_Whoops):
         result.unwrap()
 
-    assert len(history) == 4
+    assert len(history) == 3
```

Comment:
Why has this changed?

Reply:
As discussed: this is because we now store the model copy in the history rather than the original model. This test uses a model that can only be copied 3 times, meaning we now only store it three times.